pkg/config/config.go

```diff
@@ -54,6 +54,23 @@ type Config struct {
 	// documentdb.
 	// This will also change the helm chart and image names.
 	ControllerName string `json:"controller_name,omitempty"`
+	// CustomShapes defines custom structure types that can be referenced in resource fields.
+	// The outer map is keyed by shape name (e.g., "ReplicaStatus"), and the inner map
+	// is keyed by field name with string values representing field types.
+	//
+	// These shapes can be referenced in a resource's fields section using:
+	//   custom_field:
+	//     list_of: ShapeName    # For arrays of the shape
+	//     map_of: ShapeName     # For maps with the shape as values
+	//
+	// Example:
+	//   custom_shapes:
+	//     ReplicaStatus:
+	//       Status: string
+	//       Region: string
+	//       Description: string
+
```

Comment:
```suggestion
```

Reply:
https://pkg.go.dev/std is using these comments to generate documentation for the Go code.  Quoting from https://tip.golang.org/doc/comment

> “Doc comments” are comments that appear immediately before top-level package, const, func, type, and var declarations with **no intervening** newlines